### PR TITLE
Feature/purchase print

### PR DESCRIPTION
**Pull request by aberracja**

### CODE REVIEW

**Comment by aberracja** [comment]
@dancinglightning
@pjar 
@lsolniczek

File: app/assets/javascripts/print_office.js.coffee [hunk 1]
```diff
@@ -1,11 +1,15 @@
...
     $(".product_show_row_end").append html
 
 if gon.basket_id
-  jQuery.ajax(url: "/baskets/" + gon.basket_id + "/button_list.html").done (html) ->
+  $.ajax(url: "/baskets/" + gon.basket_id + "/button_list.html").done (html) ->
     $(".basket_show_end").append html
+
+if gon.purchase_id
+  $(document).on "ready" , ->
+    $(".purchase_show_last").append '<a class="btn btn-primary" href="/purchases/' + gon.purchase_id + '/invoice"> Print Invoice</a>'
```
Comment (pjar) [inline app/assets/javascripts/print_office.js.coffee:15]: Why it's done in different fashion?

File: app/controllers/orders_controller_decorator.rb [hunk 2]
```diff
@@ -1,24 +1,14 @@
 # encoding : utf-8
 OrdersController.class_eval do
 
+  before_filter :load_order, except: [:index, :new]
```
Comment (pjar) [inline app/controllers/orders_controller_decorator.rb:4]: Except should not be used - it's blacklisting and it quickly leads to miss some actions from being blacklisted. It's should be whitelisted, so only: should be used. I'm also not a fan of using before_filters and instead I would go with `@order = load_order`  in every method.

File: app/views/purchase_invoices/_invoice_footer.haml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+.row.footer
+  .col-xs-4
+    %b= raw OfficeClerk.config(:footer_left)
+    %br
+    = raw OfficeClerk.config(:footer_left2)
```
Comment (pjar) [inline app/views/purchase_invoices/_invoice_footer.haml:5]: What' s the difference between footer_left and footer_left2 ?  Names should be more descriptive

File: app/views/purchase_invoices/_invoice_footer.haml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+.row.footer
+  .col-xs-4
+    %b= raw OfficeClerk.config(:footer_left)
```
Comment (pjar) [inline app/views/purchase_invoices/_invoice_footer.haml:3]: I would go with %strong instead - it's more descriptive

File: app/views/purchase_invoices/_invoice_header.haml [hunk 5]
```diff
@@ -0,0 +1,11 @@
+.row.header
+  .col-xs-6
+    = image_tag "kauppa/invoice-logo.gif",  :id => "logo"
```
Comment (pjar) [inline app/views/purchase_invoices/_invoice_header.haml:3]: Use new hash notation

File: app/views/purchase_invoices/_invoice_items.haml [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    %td.sku.bold= t(:ean)
+    %td.desc.bold= t(:description)
+    %td.price.bold.text-right= t(:price)
+    %td.qty.bold.text-right= t(:quantity)
+    %td.total.bold.text-right= t(:total)
+  - @purchase.basket.items.each do |item|
+    %tr
+      %td.cell= item.product.ean
+      %td.cell
+        = item.product.full_name rescue item.product.name
```
Comment (pjar) [inline app/views/purchase_invoices/_invoice_items.haml:12]: This should go to some presenter / decorator.

File: app/views/purchase_invoices/_invoice_totals.haml [hunk 7]
```diff
@@ -0,0 +1,13 @@
+%table.totals
+  %tr
+    %td.cel1.bold
+      = t(:purchase_total)
+      \:
+    %td.cel2.bold{:align => "right"}= euros(@purchase.basket.total_price)
```
Comment (pjar) [inline app/views/purchase_invoices/_invoice_totals.haml:6]: Please use more descriptive names than cel1 cel2

File: app/views/receipts/show.haml [hunk 8]
```diff
@@ -0,0 +1,45 @@
+.receipt
+  #receipt-header
+    = render :partial => "orders/receipt_header"
+  .receipt-info
+    - date = @order.ordered_on || @order.ordered_on
```
Comment (pjar) [inline app/views/receipts/show.haml:5]: date should be a method in order presenter

File: print_clerk.gemspec [hunk 9]
```diff
@@ -5,7 +5,7 @@ Gem::Specification.new do |s|
   s.name        = 'print_clerk'
   s.version     = '0.4'
   s.license     = 'MIT'
-  s.summary     = 'Print invoices for a rubyclerk order'
+  s.summary     = 'Print order_invoices for a rubyclerk order'
```
Comment (pjar) [inline print_clerk.gemspec:8]: Remove underscore

File: app/views/purchase_invoices/_invoice_footer.haml (hunk 3 above)
Comment (rubydesign) [inline app/views/purchase_invoices/_invoice_footer.haml:5]: @pjar nice of you to go through this: we spoke today that they would redo the pr with only purchase functionality. (not change anything that is essentially working)
These templates are copied and while i said to copy the order stuff, i thought as a start, not to check in.

File: app/views/purchase_invoices/_invoice_header.haml (hunk 5 above)
Comment (rubydesign) [inline app/views/purchase_invoices/_invoice_header.haml:3]: I am btw an old notation guy, i find it more descriptive

**Pull request closed without merging**